elixir/apps/web/lib/web/components/form_components.ex

```diff
@@ -580,4 +581,16 @@ defmodule Web.FormComponents do
 
     [icon_size[size], spacing[size]]
   end
+
+  # The phx-disable-with attribute on submit buttons only lasts until the button press is
+  # acknowledged by the server. After that, it's restored, even if the form itself is still
+  # submitting. This is a problem when the form is slow to submit, as the user can press the
+  # button again, causing the form to submit twice. This happens on auth pages where the submission
+  # is synchronous and takes a constant time. So we use custom JS to disable the button instead,
+  # fixing UX.
+  # This is only useful for synchronous form submissions.
+  def disable_button(js \\ %JS{}, id) do
+    js
+    |> JS.add_class("opacity-75 cursor-wait", to: "##{id}")
+  end
```

Comment:
Yeah, I thought that might be the case too, but the only difference I've found between forms this works for and forms it doesn't is whether they're synchronous.

I'll take another look into it this morning.

Reply:
We should not bind this to button's `phx-click` but to `phx-submit` of the form.

Reply:
I'm not phx-disable-with is designed to work with synchronous submissions? The "server response" in this case happens immediately because the handler is synchronous.

Reply:
On that note, was there a specific reason we aren't using liveview navigation for these? IIRC it's to handle the user refreshing on the sign in pages but not sure.

Reply:
This has been updated to use `phx-submit` and a new hook to disable the submit button, with a reset back to normal state just in case the request fails or something.